### DISCUSSION TITLE
Will you be able to change the DAC_CH?

### DISCUSSION THREAD

**Discussion by chrmlinux** [Roger-random/ESP_8_BIT_composite#33] [Ideas]
As you know, there are two DAC_CHs on the ESP32.
However, the output of ESP_8_BIT_composite is fixed to CH1.

ex)
dac_output_enable(DAC_CHANNEL_1); // DAC, video on GPIO25

Is it possible to change the CH to switch between DAC_CHANNEL_1 and DAC_CHANNEL_2 instead of fixed?
Here is what I modified
https://github.com/chrmlinux/ESPDisplay/

**Comment by Roger-random** [comment]
Possible, but at the moment unlikely. The goal of this particular library is to make composite video easily accessible to beginners, keeping the number of options low to minimize confusion.

I expect developers who are advanced enough to want to switch between DAC_CHANNEL_1 and DAC_CHANNEL_2 will also be advanced enough to modify the library to suit their needs. Which I understand is exactly what you have done!

However, this has not been the only request for advanced feature. I'm trying to figure out how to keep things beginner-friendly while still offering power for advanced users, but nothing well formed yet to announce.

**Reply by chrmlinux** [reply]
Thank you for your reply
Because you got the MIT license
I will include it by saying C_ESP_8_BIT_composite.
Furthermore, registration as Arduino Library has been completed.
Please search from the library management with "ESPDisplay".
Currently, we are remodeling to work with ESPC3U etc. by a method that does not use DAC.
Thank you for your consideration.

https://twitter.com/chrmlinux03/status/1513084933266640896?s=20&t=ZEabsV7b92xqaeagevGfeQ

**Reply by chrmlinux** [reply]
This mission started from the time ATOMDisplay was released, just transferring from the LovyanFGX sprite framebuffer to the ESP_8_BIT_composite frame.
It works with a simple configuration
https://qiita-com.translate.goog/chrmlinux03/items/c66ca9ff5abcc9560902?_x_tr_sl=ja&_x_tr_tl=en&_x_tr_hl=ja&_x_tr_pto=wapp

**Reply by Roger-random** [reply]
I don't have the same hardware to play with it myself, but from your posts it looks pretty good!

**Reply by chrmlinux** [reply]
Thank you for your reply
Currently, the operation has been confirmed with M5StickC / M5Stack / M5AtomLite / M5picoD4 (G25 / G26 is an open device).
Since the FPS of ESP_8_BIT_composite is 60Hz, it seems that the display will be in time even if a fairly complicated process is incorporated in the loop.

I think this is a good encounter
We are looking forward to hearing from you
ENjoy ESP32 Life!
